### PR TITLE
Slow bundle dialog

### PR DESCRIPTION
**Pull request by pkriens**
Typing in the filter box did a slow update. This PR uses a (copy of) the FilteredTree that is supposed to be much faster. Now also supports wildcards a la globbing

---
 Signed-off-by: Peter Kriens <Peter.Kriens@aQute.biz>

### CODE REVIEW

**Comment by pkriens** [comment]
#5699

File: bndtools.core/src/bndtools/editor/project/AvailableBundlesPart.java [hunk 1]
```diff
@@ -16,89 +14,65 @@
...
+																		public boolean select(Viewer viewer,
+																			Object parentElement, Object element) {
+																			boolean select = false;
+																			if (element instanceof RepositoryBundle) {
+																				RepositoryBundle repoBundle = (RepositoryBundle) element;
+																				RepositoryPlugin repo = repoBundle
+																					.getRepo();
+
+																				if (includedRepos == null) {
 																					select = true;
```
Comment (kriegfrj) [inline bndtools.core/src/bndtools/editor/project/AvailableBundlesPart.java:60]: Sorry to be painful but it's difficult to review when there's these unrelated whitespace-only changes in the commit. 😦

Comment (kriegfrj) [reply bndtools.core/src/bndtools/editor/project/AvailableBundlesPart.java:60]: (Thanks for taking it on!)

Comment (pkriens) [reply bndtools.core/src/bndtools/editor/project/AvailableBundlesPart.java:60]: I've no idea how it can happen. We automatically format on save and the formatting instructions have not changed. Somehow Eclipse has some randomness in it. I agree it is a pain.

Comment (kriegfrj) [reply bndtools.core/src/bndtools/editor/project/AvailableBundlesPart.java:60]: Well, look what I found: 
Github diff

I'll get to reviewing it then!

**Comment by kriegfrj** [comment]
It looks good to me on the surface of it. I have tried it in my local build and it seems to run noticeably faster even on reasonably small repository bases. 👏 🥳  Ideally it would be good also for this to have some time in pre-release so that the community can put it through its paces before we release, as it touches a fairly core component (though there is always the text file workaround if the GUI fails for some reason).

I have not analyzed the code in detail (well, actually I started and then I read it properly and realised you'd lifted it from Eclipse, so I didn't bother reviewing further. :smile:). A few questions/comments: 
* What was your reasoning behind copying the code from Eclipse rather than referencing it directly? Is this component not available in our current baseline of Eclipse? If so, perhaps it is time to consider lifting the baseline, given that we've already gone from Java 8 to Java 17 in this release?
* If we're going to stick with a copy rather than a reference:
  * Can we document why this decision was made in the commit message and here?
  * Can we add/alter comments to/at the top of the copied files to make it clear at a glance that they are copies (including a reference to the FQN of the original class)?

**Comment by pkriens** [comment]
I copied the code from Eclipse because the filtering cross all levels. It was not possible to only filter on bsn. I spent 2 hours or so because there are options that indicate what gets filtered/is visible etc but in the end I had to make a code change. The standard filtering was akward, it would hide all the versions and no way to open a bsn to see the versions. I tried sub classing but this dragged in 60% of the code.

It is not very big so I do not mind that much.

I can add a reference. I guess it was obvious :-)

**Comment by kriegfrj** [comment]
> I can add a reference. I guess it was obvious :-)

I must be a bit slow or inattentive because I didn't notice at first and wasted a bunch of time reviewing what I thought was your code when it was actually Eclipse's code. :smile: Though it was obvious in hindsight...

**Comment by pkriens** [comment]
My bad ... excuses.

**Comment by kriegfrj** [comment]
> ... excuses.

Ditto for me... sorry, that was meant to be a jab at myself, not at you...
